### PR TITLE
WIP feat: Support go std logger

### LINKED ISSUE #245: Log package: support logging to file
/kind feature

### LINKED ISSUE #291: How to write log to file?
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind feature

**What you expected to happen**:

How to write log to a file?

### PR DESCRIPTION
**Pull request by gaocegege**
Signed-off-by: Ce Gao <gaoce@caicloud.io>

**What this PR does / why we need it**:

Add your description

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #245 

Fixes #291 

Reference to #

**Special notes for your reviewer**:

/cc @your-reviewer

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by gaocegege** [comment]
Here is an example

```go
func main() {
	// Print nirvana banner.
	fmt.Println(nirvana.Logo, nirvana.Banner)

	// Create nirvana command.
	cmd := config.NewNamedNirvanaCommand("server", config.NewDefaultOption())

	// Create plugin options.
	metricsOption := metrics.NewDefaultOption() // Metrics plugin.
	reqlogOption := reqlog.NewDefaultOption()   // Request log plugin.
	versionOption := pversion.NewOption(        // Version plugin.
		"helloworld",
		version.Version,
		version.Commit,
		version.Package,
	)

	// Enable plugins.
	cmd.EnablePlugin(metricsOption, reqlogOption, versionOption)

	// Create server config.
	serverConfig := nirvana.NewConfig()

	f, err := os.OpenFile("testlogfile", os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666)
	if err != nil {
		fmt.Printf("Failed to open the file: %v", err)
		os.Exit(1)
	}
	defer f.Close()

	// Configure APIs. These configurations may be changed by plugins.
	serverConfig.Configure(
		nirvana.Logger(log.NewGoStandardLogger(0, f, "test", stdlog.LstdFlags)),
		nirvana.Filter(filters.Filters()...),
		nirvana.Modifier(modifiers.Modifiers()...),
		nirvana.Descriptor(apis.Descriptor()),
	)

	// Set nirvana command hooks.
	cmd.SetHook(&config.NirvanaCommandHookFunc{
		PreServeFunc: func(config *nirvana.Config, server nirvana.Server) error {
			// Output project information.
			config.Logger().Infof("Package:%s Version:%s Commit:%s", version.Package, version.Version, version.Commit)
			return nil
		},
	})

	// Start with server config.
	if err := cmd.ExecuteWithConfig(serverConfig); err != nil {
		serverConfig.Logger().Fatal(err)
	}
}
```

**Comment by gaocegege** [comment]
Test cases will be added this or next week.

File: log/golog.go [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+// goLogger is the logger with go logger.
+type goLogger struct {
+	l         *log.Logger
+	level     Level
+	calldepth int
+}
+
+// NewGoStandardLogger creates a zap logger.
+func NewGoStandardLogger(level Level, out io.Writer, prefix string, flag int) Logger {
```
Comment (gaocegege) [inline log/golog.go]: Not sure if we should provide a NewDefaultGoStandardLogger since we already have one using stdLogger.

/cc @kdada

Comment (kdada) [reply log/golog.go]: Why do you wrap a standard log rather than output data to the `out` directly?

Comment (kdada) [reply log/golog.go]: The flag `stdlog.LstdFlags` is duplicated with the result of `prefix()`

Comment (gaocegege) [reply log/golog.go]: In my opinion, if there is a mature tool, prefer not to reinvent a new one. And std log supports log header line or some other features. WDYT

Comment (gaocegege) [reply log/golog.go]: > The flag `stdlog.LstdFlags` is duplicated with the result of `prefix()`

I think the prefix and stdFlags serves for different purposes. But it also works for me if we remove the support for prefix or flags.

**Comment by kdada** [comment]
@gaocegege Could you fix the CI?

**Comment by gaocegege** [comment]
Sure. If the approach LGTM, I can fix it soon.

File: log/golog.go (hunk 1 above)
Comment (kdada) [reply log/golog.go]: I think the behaviors of this logger should be splitted into two scenarios:
1. A standard implementation for std log.
2. A pure file logger for the log format of `prefix()`. 

So I think you can remove the `prefix()` from this PR and make it be the 1st implementation.

Comment (kdada) [reply log/golog.go]: Based on my personal preference, I'd like the 2nd implementation.

Comment (gaocegege) [reply log/golog.go]: SGTM

Comment (gaocegege) [reply log/golog.go]: BTW, can we output the log to file and to STDOUT? Maybe register two loggers.

**Comment by gaocegege** [comment]
@kdada PTAL

Should we implement the corresponding logic for log plugin in this PR?

**Comment by kdada** [comment]
> Should we implement the corresponding logic for log plugin in this PR?

Yes. It makes the logger easy to use.

**Comment by hezhizhen** [comment]
Any updates? @kdada @gaocegege

**Comment by gaocegege** [comment]
@hezhizhen Recently working on Clever and do not have the bandwidth to deal with it.

@kdada Can we merge it if we implement the log plugin in another PR?

**Comment by ddysher** [comment]
@iawia002 please chime in and move this forward (either check it in or close it).

**Comment by gaocegege** [comment]
/close

Do not have the bandwidth to maintain it.

**Pull request closed without merging**